xbmc/guilib/GUIAudioManager.cpp

```diff
@@ -295,9 +270,9 @@ bool CGUIAudioManager::Load()
       if (id != ACTION_NONE && !strFile.empty())
       {
         std::string filename = URIUtils::AddFileToFolder(m_strMediaDir, strFile);
-        IAESound *sound = LoadSound(filename);
+        auto sound = LoadSound(filename);
         if (sound)
-          m_actionSoundMap.insert(std::pair<int, IAESound *>(id, sound));
+          m_actionSoundMap.emplace(id, std::move(sound));
```

Comment:
do we benefit from the `std::move` ? i believe we will pass to the constructor `std::pair( const T1& x, const T2& y );` anyway because of the `id` variable. but i might be wrong

Reply:
Good question. Going by [cppreference](https://en.cppreference.com/w/cpp/container/map/emplace) it should work as intended:

> ... The constructor of the new element (i.e. std::pair<const Key, T>) is called with exactly the same arguments as supplied to emplace, forwarded via std::forward<Args>(args)....

So the constructor of `std::shared_ptr` in the `std::map` should be called with the rvalue reference and a move should take place.

Reply:
ok, looks you're right.. this quick test reveals it. honestly i didn't expect

```c++
std::string key = "key1"; std::string val = "This is the value";
    
std::map<std::string, std::string> myMap;
myMap.emplace(key, std::move(val)); // do we move from val ?
        
std::cout << "key: " << key << std::endl;
std::cout << "value: " << val << std::endl; // yes we do!
```